src/cargo/ops/cargo_install.rs

```diff
@@ -345,7 +388,7 @@ fn install_one(
             pkg
         );
         config.shell().status("Ignored", &msg)?;
-        return Ok(false);
+        return Ok(());
     }
 
     config.shell().status("Installing", &pkg)?;
```

Comment:
From a PR like this my naive expectation was that everything prior to this line would happen in a "serial batch" where the build afterwards would happen in parallel (eventually). Would it be possible to move all the pkg-specific validation above into the `determine_package` function?

Reply:
I gave it a shot! Most of it was able to move over. A couple things had to stay - which I'll leave notes for inline. I did it in a separate commit "Refactor additional validation steps to determine_package" so it's easier to review (and remove if we decide against it).